### PR TITLE
Test: e2e test for teams

### PR DESCRIPTION
**Pull request by Ajen07**
# Description

- Added e2e tests for Teams controller

## Issue link

[Fixes # (issue)](https://app.clickup.com/t/86azr0q4g)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested with `yarn test:docker`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: test/teams.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        app.use(cookieParser());
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await prisma.$disconnect();
+        await app.close();
+    });
+
+    describe("GET /teams - gets all voyage teams", () => {
```
Comment (cherylli) [inline test/teams.e2e-spec.ts:37]: could you please add a test for non admin user access for this route? (should return 403)
Looks like it's also not on swagger, if you don't mind could you add that to swagger as well?

File: test/teams.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        });
+        it("should return 401 when user is not logged in", async () => {
+            await request(app.getHttpServer())
+                .get("/teams")
+                .set("Authorization", `Bearer ${undefined}`)
+                .expect(401)
+                .expect("Content-Type", /json/);
+        });
+    });
+    describe("GET /teams/voyages/:voyageid - Gets all team for a voyage given a voyage id", () => {
```
Comment (cherylli) [inline test/teams.e2e-spec.ts:65]: this route also needs a 403 for non admin, same as above

**Review by cherylli — COMMENTED** [review]
looks good, just need to add 403 tests for non admin users for some routes

File: test/teams.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    beforeAll(async () => {
+        const moduleFixture: TestingModule = await Test.createTestingModule({
+            imports: [AppModule],
+        }).compile();
+
+        await seed();
+        app = moduleFixture.createNestApplication();
+        prisma = moduleFixture.get<PrismaService>(PrismaService);
+        app.useGlobalPipes(new ValidationPipe());
+        app.use(cookieParser());
```
Comment (timDeHof) [inline test/teams.e2e-spec.ts:28]: Should we add an `await` before `moduleFixture.createNestApplication()`,`app.useGlobalPipes(new ValidationPipe())`, and `app.use(cookieParser())` to ensure that they complete before the application is initialized?

File: test/teams.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+import * as cookieParser from "cookie-parser";
+
+//Logged in user is Jessica Williamson for admin routes /teams and /teams/voyages/:voyageid
+//Logged in user is Dan ko for team member routes /teams/:teamid and /teams/:teamid/members
+//Dan Ko is part of the team with team id 4
+
+describe("Teams Controller (e2e)", () => {
+    let app: INestApplication;
+    let prisma: PrismaService;
+    let accessToken: any;
```
Comment (timDeHof) [inline test/teams.e2e-spec.ts:17]: maybe define the accessToken as a string to prevent potential runtime errors.

**Review by timDeHof — CHANGES_REQUESTED** [review]
All tests passed. The test file is well-structured and follows good practices for e2e testing in Nestjs. However, I added a few suggestions to the code that could improve the code quality.

File: test/teams.e2e-spec.ts (hunk 3 above)
Comment (cherylli) [reply test/teams.e2e-spec.ts:28]: are they async functions? I've actually never seen them used with await even in the nestjs docs

File: test/teams.e2e-spec.ts (hunk 1 above)
Comment (Ajen07) [reply test/teams.e2e-spec.ts:37]: Ok , I will add the test for non admin user and update the swagger responses as well.

File: test/teams.e2e-spec.ts (hunk 3 above)
Comment (Ajen07) [reply test/teams.e2e-spec.ts:28]: We don't need to add await  as they are not promises , they are all synchronous operations and `app.use(cookieParser())` is a middleware for parsing cookies in our http request for all endpoints.

**Review by timDeHof — APPROVED** [review]
Looks good, All tests passed

**Review by JoshuaHinman — DISMISSED** [review]
All tests passed and code looks great

**Review by cherylli — APPROVED** [review]
ran all the tests, all good. Thanks for adding the 403

**Review by timDeHof — APPROVED** [review]
All the tests passed!

**Pull request merged**